src/plugins/unified_search/public/filter_bar/filter_editor/filter_editor.tsx

```diff
@@ -143,42 +144,70 @@ export interface FilterEditorComponentProps {
   suggestionsAbstraction?: SuggestionsAbstraction;
   docLinks: DocLinksStart;
   filtersCount?: number;
+  dataViews: DataViewsContract;
 }
 
 export type FilterEditorProps = WithEuiThemeProps & FilterEditorComponentProps;
 
 interface State {
+  indexPatterns: DataView[];
   selectedDataView?: DataView;
   customLabel: string | null;
   queryDsl: string;
   isCustomEditorOpen: boolean;
   localFilter: Filter;
+  isLoadingDataView?: boolean;
 }
 
 class FilterEditorComponent extends Component<FilterEditorProps, State> {
   constructor(props: FilterEditorProps) {
     super(props);
-    const dataView = this.getIndexPatternFromFilter();
+    const dataView = getIndexPatternFromFilter(props.filter, props.indexPatterns);
     this.state = {
+      indexPatterns: props.indexPatterns,
       selectedDataView: dataView,
       customLabel: props.filter.meta.alias || '',
-      queryDsl: this.parseFilterToQueryDsl(props.filter),
+      queryDsl: this.parseFilterToQueryDsl(props.filter, props.indexPatterns),
       isCustomEditorOpen: this.isUnknownFilterType() || !!this.props.filter?.meta.isMultiIndex,
       localFilter: dataView ? merge({}, props.filter) : buildEmptyFilter(false),
+      isLoadingDataView: !Boolean(dataView),
     };
   }
 
   componentDidMount() {
-    const { localFilter, queryDsl, customLabel } = this.state;
+    const { localFilter, queryDsl, customLabel, selectedDataView } = this.state;
     this.props.onLocalFilterCreate?.({
       filter: localFilter,
       queryDslFilter: { queryDsl, customLabel },
     });
     this.props.onLocalFilterUpdate?.(localFilter);
+    if (!selectedDataView) {
+      const dataViewId = this.props.filter.meta.index;
+      if (!dataViewId) {
+        this.setState({ isLoadingDataView: false });
+      } else {
+        this.props.dataViews
+          .get(dataViewId)
+          .then((dataView) => {
+            this.setState({
+              selectedDataView: dataView,
+              isLoadingDataView: false,
+              indexPatterns: [dataView, ...this.props.indexPatterns],
+              localFilter: merge({}, this.props.filter),
+              queryDsl: this.parseFilterToQueryDsl(this.props.filter, this.state.indexPatterns),
+            });
+          })
+          .catch(() => {
+            this.setState({
+              isLoadingDataView: false,
+            });
+          });
+      }
+    }
```

Comment:
Not that this is the biggest change of this PR, in case the selectedDataView can't be found in the provided list of DataViews, it's attempting to load the data view based on the id in the filter settings. it's worth to consider displaying an error message if this doesn't work.